tests/test_auth.py

```diff
@@ -0,0 +1,262 @@
+    mock_signing_key = MagicMock()
+    mock_jwks_client.get_signing_key_from_jwt.return_value = mock_signing_key
+
+    signing_key = handler._fetch_signing_key("https://example.com/jwks_uri", "fake_token")
+    assert signing_key == mock_signing_key
+    handler.jwks_client_factory.assert_called_with("https://example.com/jwks_uri")
+    mock_jwks_client.get_signing_key_from_jwt.assert_called_with("fake_token")
+
+
+def test_decode_jwt_success(handler):
+    decoded_token = {
+        "aud": "zenith-jupyter",
+        "exp": 12345,
+        "iss": "https://example.com",
+        "iat": 12344,
+        "short_name": "user",
+        "projects": {},
+    }
+    mock_signing_key = MagicMock()
+    mock_signing_key.key = "fake_key"
+
+    with patch("jwt.decode", return_value=decoded_token):
+        result = handler._decode_jwt("fake_token", mock_signing_key, ["RS256"])
+        assert result == decoded_token
+
+
+def test_decode_jwt_failure(handler):
+    mock_signing_key = MagicMock()
+    mock_signing_key.key = "fake_key"
+
+    with patch("jwt.decode", side_effect=jwt.InvalidTokenError("Invalid token")):
+        with pytest.raises(HTTPError) as exc_info:
+            handler._decode_jwt("fake_token", mock_signing_key, ["RS256"])
+        assert exc_info.value.status_code == 401
+
+
+def test_normalize_projects_invalid_json(handler):
+    decoded_token = {"projects": "invalid_json"}
+    result = handler._normalize_projects(decoded_token)
+    assert result == {}  # Expect an empty dict instead of a raw string
+
+
+def test_normalize_projects_none(handler):
+    decoded_token = {}
+    result = handler._normalize_projects(decoded_token)
+    assert result == {}
+
+
+@pytest.mark.asyncio
+async def test_get():
+    # Create a real Application instance with necessary settings
+    application = Application()
+    application.settings = {
+        "hub": MagicMock(base_url="/hub/"),  # Mock 'hub' with base_url as a string
+        "cookie_secret": b"secret",  # Add other required settings
+    }
+
+    # Mock request with a connection attribute
+    request = MagicMock(spec=HTTPServerRequest)
+    request.connection = MagicMock()  # Add the 'connection' attribute
+
+    # Create an instance of the handler
+    handler = BricsLoginHandler(application, request, oidc_server="https://example.com")
+
+    # Mock handler dependencies
+    handler._extract_token = MagicMock(return_value="mock_token")
+    handler._fetch_oidc_config = AsyncMock(
+        return_value={"id_token_signing_alg_values_supported": ["RS256"], "jwks_uri": "https://example.com/jwks"}
+    )
+    handler._parse_oidc_config = MagicMock(return_value=(["RS256"], "https://example.com/jwks"))
+    handler._fetch_signing_key = MagicMock(return_value=MagicMock(key="mock_key"))
+    handler._decode_jwt = MagicMock(return_value={"short_name": "test_user", "projects": {}})
+    handler._normalize_projects = MagicMock(return_value={"project1": "value1"})
```

Comment:
I think the return value of this function should be a dict of lists

```suggestion
    handler._normalize_projects = MagicMock(return_value={"project1": ["value1"]})
```